File: ecommerce-web/src/app/core/models/order.ts
export interface OrderItem {
  id: number;
  productId: number;
  productName?: string;
  quantity: number;
  price: number;
}

export interface OrderGiftSummary {
  giftId: number;
  giftName: string;
  quantity: number;
  giftRuleId: number;
}

export interface Order {
  id: number;
  customerId: number;
  subtotalAmount?: number;
  taxAmount?: number;
  shippingAmount?: number;
  totalAmount: number;
  createdAt: string;
  status: string;
  promotionCode?: string | null;
  orderItems: OrderItem[];
  assignedGifts?: OrderGiftSummary[];
}

export interface CreateOrderPayload {
  items: { productId: number; quantity: number }[];
  promotionCode?: string;
}

File: EcommerceAPI/Models/Order.cs
namespace EcommerceAPI.Models;

public class Order
{
    public int Id { get; set; }
    public int CustomerId { get; set; }
    public decimal SubtotalAmount { get; set; }
    public decimal TaxAmount { get; set; }
    public decimal ShippingAmount { get; set; }
    public decimal TotalAmount { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public string Status { get; set; } = "Pending";

    /// <summary>Optional promotion code supplied at checkout; used when confirming payment for gift rules.</summary>
    public string? PromotionCode { get; set; }

    // Navigation properties
    public Customer? Customer { get; set; }
    public ICollection<OrderItem> OrderItems { get; set; } = new List<OrderItem>();
    public ICollection<OrderGift> OrderGifts { get; set; } = new List<OrderGift>();
}